protocols/kad/src/behaviour.rs

```diff
@@ -440,15 +443,17 @@ where
             protocol_config: config.protocol_config,
             record_filtering: config.record_filtering,
             queued_events: VecDeque::with_capacity(config.query_config.replication_factor.get()),
+            listen_addresses: Default::default(),
             queries: QueryPool::new(config.query_config),
             connected_peers: Default::default(),
             add_provider_job,
             put_record_job,
             record_ttl: config.record_ttl,
             provider_record_ttl: config.provider_record_ttl,
             connection_idle_timeout: config.connection_idle_timeout,
-            local_addrs: HashSet::new(),
             caching: config.caching,
+            external_addresses: ExternalAddresses::with_limit(MAX_LOCAL_EXTERNAL_ADDRS),
```

Comment:
Should we make `MAX_LOCAL_EXTERNAL_ADDRS` the default for `ExternalAddresses`? I don't see how `MAX_LOCAL_EXTERNAL_ADDRS` is specific to Kademlia.